src/main/java/xyz/nkomarn/harbor/util/PlayerManager.java

```diff
@@ -62,73 +73,51 @@ public void clearCooldowns() {
      * Checks if a player is considered "AFK" for Harbor's player checks.
      *
      * @param player The player to check.
+     *
      * @return Whether the player is considered AFK.
      */
     public boolean isAfk(@NotNull Player player) {
-        if (!harbor.getConfiguration().getBoolean("afk-detection.enabled")) {
-            return false;
-        }
-
-        Optional<Essentials> essentials = harbor.getEssentials();
-        if (essentials.isPresent()) {
-            User user = essentials.get().getUser(player);
-
-            if (user != null) {
-                return user.isAfk();
-            }
-        }
-
-        if (!playerActivity.containsKey(player.getUniqueId())) {
-            return false;
+        // If there are no providers registered, we go to the default provider
+        if(oredProviders.isEmpty() && andedProviders.isEmpty()){
+            return defaultProvider.isAFK(player);
         }
+        return oredProviders.stream().anyMatch(provider -> provider.isAFK(player)) ||
+                (!andedProviders.isEmpty() && andedProviders.stream().allMatch(provider -> provider.isAFK(player)));
+    }
 
-        long minutes = TimeUnit.MILLISECONDS.toMinutes(System.currentTimeMillis() - playerActivity.get(player.getUniqueId()));
-        return minutes >= harbor.getConfiguration().getInteger("afk-detection.timeout");
+    @EventHandler
+    public void onQuit(PlayerQuitEvent event) {
+        UUID uuid = event.getPlayer().getUniqueId();
+        cooldowns.remove(uuid);
     }
 
+
     /**
-     * Sets the given player's last activity to the current timestamp.
+     * Add an AFK Provider to harbor, so an external plugin can provide an AFK status to harbor
      *
-     * @param player The player to update.
+     * @param provider  The {@link AFKProvider} to be added
+     * @param logicType The type of logic (And or Or, {@link LogicType}) to be used with the given provider
      */
-    public void updateActivity(@NotNull Player player) {
-        playerActivity.put(player.getUniqueId(), System.currentTimeMillis());
+    public void addAfkProvider(AFKProvider provider, LogicType logicType) {
+        (logicType == LogicType.AND ? andedProviders : oredProviders).add(provider);
```

Comment:
Bad readability a normal if else or switch would be better.

Not that big of a deal, but just so you know ig. 😅

Reply:
8PM Me thought that was more readable than using a hashmap to deal with it, but an if/else would be more readable (though in this case, I'm going to go with a switch)